include/simdjson/common_defs.h

```diff
@@ -121,10 +121,29 @@ constexpr size_t DEFAULT_MAX_DEPTH = 1024;
 
 #endif // MSC_VER
 
+
+// some systems have string_view even if SIMDJSON_CPLUSPLUS17 is 0,
```

Comment:
I'm *absolutely not* asking you to change this right now--there would be a lot of testing involved--but I am wondering if `#if __cpp_lib_string_view` will do the trick here (see https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros#C.2B.2B11).

Reply:
Let me try. I am certainly eager to simplify.

Reply:
I tried, but it does not work. The official clang defines __cpp_lib_string_view but not the Apple clang.

Reply:
I will document it.

Reply:
Note that I have added a check for __cpp_lib_string_view